src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -692,6 +752,13 @@ export class EditorPart extends Part implements IEditorPart {
 		return fallback;
 	}
 
+	private onWillGroupsChange() {
```

Comment:
Not a big fan of sprinkling this method through places here, in most cases I believe relying on events such as active editor change or visible editor change should be enough? On top of that, there are many group related events: onDidChangeActiveGroup, onDidAddGroup, onDidRemoveGroup, onDidMoveGroup, onDidActivateGroup

Reply:
We can not use the events as they are all triggered after a change has been done. We have to exit the maximize state before we do any changes to the groups.